src/sources/text-document.ts

```diff
@@ -0,0 +1,25 @@
+import { Document, IDataSource } from '../types';
+
+export const NAME = 'text' as const;
+
+export type TextDocumentOptions = {
+  content: string;
+  filename: string;
```

Comment:
This is currently assuming "text" will always be a file (using file:// protocol). Can we make this class take a `url` instead of filename and then it's more generally applicable?

Reply:
Yeah OK